modules/products/bitbucket/variables.tf

```diff
@@ -0,0 +1,81 @@
+variable "environment_name" {
+  description = "Name of the environment."
+  type        = string
+  validation {
+    condition     = can(regex("^[a-zA-Z][a-zA-Z0-9\\-]{1,24}$", var.environment_name))
+    error_message = "Invalid environment name. Valid name is up to 25 characters starting with alphabet and followed by alphanumerics. '-' is allowed as well."
+  }
+}
+
+variable "namespace" {
+  description = "The namespace where Bitbucket pod will be installed."
+  type        = string
+}
+
+variable "vpc" {
+  description = "VPC module that hosts the products."
+  type        = any
+}
+
+variable "eks" {
+  description = "EKS module that hosts the product."
+  type        = any
+}
+
+variable "ingress" {
+  default = null
+  type    = any
+}
+
+variable "db_major_engine_version" {
+  description = "The database major version to use."
+  type        = string
+}
+
+variable "db_allocated_storage" {
+  description = "Allocated storage for database instance in GiB."
+  type        = number
+}
+
+variable "db_instance_class" {
+  description = "Instance class of the RDS instance."
+  type        = string
+}
+
+variable "db_iops" {
+  description = "The requested number of I/O operations per second that the DB instance can support."
+  type        = number
+}
+
+variable "bitbucket_configuration" {
+  description = "Bitbucket resource spec and chart version"
+  type        = map(any)
+  validation {
+    condition = (length(var.bitbucket_configuration) == 6 &&
+      alltrue([
+        for o in keys(var.bitbucket_configuration) : contains([
+          "helm_version", "cpu", "mem", "min_heap", "max_heap", "license"
+        ], o)
+    ]))
+    error_message = "Bitbucket configuration is not valid."
+  }
+}
+
+variable "pvc_claim_name" {
+  description = "Persistent volume claim name for shared home."
+  type        = string
+  validation {
+    condition     = can(regex("^[a-zA-Z]+[a-zA-Z0-9|\\-|_]*$", var.pvc_claim_name))
+    error_message = "Invalid claim name."
+  }
+}
+
+variable "admin_configuration" {
+  description = "Bitbucket admin configuration"
+  type        = map(any)
+  validation {
+    condition = (length(var.admin_configuration) == 4 &&
```

Comment:
Does it mean it is now mandatory? It is probably fine to have it mandatory, I just want to highlight that Bitbucket doesn't have the same limitation as Bamboo (which needs to be fully configured if using unattended setup).

Reply:
Good point!

Reply:
No, they are not mandatory, because the input values come from root level and all have default value there. The same for Bamboo.

Reply:
As discussed, created [TFKUBE-330](https://bulldog.internal.atlassian.com/browse/TFKUBE-330).